File: FrozenFramework/src/main/java/lakeshire/github/com/frozenframework/fragment/AbsPageHolderFragment.java
package lakeshire.github.com.frozenframework.fragment;

import android.support.v4.view.ViewPager;
import android.view.View;

import java.util.List;

import in.srain.cube.views.ptr.PtrFrameLayout;
import lakeshire.github.com.frozenframework.R;
import lakeshire.github.com.frozenframework.adapter.PageFragmentAdapter;

/**
 * 有子页面的Fragment抽象类
 *
 * @author lakeshire
 */
abstract public class AbsPageHolderFragment extends BasePullFragment {
    protected ViewPager mPager;
    protected PageFragmentAdapter mAdapter;

    @Override
    abstract public int getContainerLayoutId();

    @Override
    public void initUi() {
        super.initUi();

        mPager = findView(R.id.pager);
        List<IPager> fragments = initFragments();
        mAdapter = new PageFragmentAdapter(fragments, getChildFragmentManager());
        mPager.setAdapter(mAdapter);
    }

    abstract protected List<IPager> initFragments();

    @Override
    public void onRefresh() {
        refreshComplete();
    }

    @Override
    public boolean checkCanRefresh(PtrFrameLayout frame, View content, View header) {
        return false;
    }
}

File: FrozenFramework/src/main/java/lakeshire/github/com/frozenframework/fragment/AbsPageHolderFragment.kt
package lakeshire.github.com.frozenframework.fragment

import `in`.srain.cube.views.ptr.PtrFrameLayout
import android.support.v4.view.ViewPager
import android.view.View
import lakeshire.github.com.frozenframework.R
import lakeshire.github.com.frozenframework.adapter.PageFragmentAdapter

/**
 * 有子页面的Fragment抽象类

 * @author lakeshire
 */
abstract class AbsPageHolderFragment : BasePullFragment() {
    lateinit var mPager: ViewPager
    lateinit var mAdapter: PageFragmentAdapter

    abstract override val containerLayoutId: Int

    override fun initUi() {
        super.initUi()

        mPager = mContainerView.findViewById(R.id.pager)
        val fragments = initFragments()
        mAdapter = PageFragmentAdapter(fragments, childFragmentManager)
        mPager.adapter = mAdapter
    }

    abstract fun initFragments(): List<IPager>

    override fun onRefresh() {
        refreshComplete()
    }

    override fun checkCanRefresh(frame: PtrFrameLayout, content: View, header: View): Boolean {
        return false
    }
}
